### ISSUE TITLE
ModuleNotFoundError: No module named 'kale

### ISSUE THREAD

**Issue by chetanoruganti** [kubeflow-kale/kale#46]
Is it mandatory to install kale in the images which we are tagging to the cells of ipynb .
As i see below code embedded into the functions which are mapped to the comp.func_to_container_op functions .

           from kale.utils import pod_utils
           from kale.marshal import resource_save as _kale_resource_save
           from kale.marshal import resource_load as _kale_resource_load

**Comment by StefanoFioravanzo** [comment]
Hello @chetanoruganti . Yes, every pipeline step should have Kale installed. Please note that, if you are on MiniKF, you can take advantage of the Rok's snapshotting functionality. When running Kale, if you select the `Use this notebook's volumes` toggle, Rok will snapshot the current environment and use the NotebookServer image to run the Pipeline steps.

Since Kale is already installed in the NotebookServer image you don't have to build your own images

**Comment by chetanoruganti** [comment]
Hello @StefanoFioravanzo . Thanks you for quick response
In some scenarios if we want to use different docker images for different pipeline steps can we achieve it .

**Comment by StefanoFioravanzo** [comment]
Yes, currently Kale lets you specify a single docker image for all pipeline steps. Just wondering, where are you running Kale? Your own cluster? If you try it out using MiniKF you can take advantage of the integration with Rok, that will take snapshots of the working environment for you so you don't have to build images by hand

**Comment by chetanoruganti** [comment]
@StefanoFioravanzo  yes running kale on local cluster .
